### PR TITLE
Enable pins

### PR DESCRIPTION
**Pull request by malikabhi05**
Initial commit to enable GPIO, AIO, PWM in as many pins as possible and add guards for I2C and UART.

### CODE REVIEW

File: source/aio.c [hunk 1]
```diff
@@ -61,6 +61,43 @@ mraa_aio_init(unsigned int pin)
         return NULL;
     }
 
+    struct device* pinmux_dev = device_get_binding(CONFIG_PINMUX_DEV_NAME);
+    if (pinmux_dev == NULL) {
+        printf("Failed to get binding for pinmux\n");
+    }
```
Comment (jontrulson) [inline source/aio.c]: Should there be a return NULL here?

File: source/gpio.c [hunk 2]
```diff
@@ -82,6 +83,43 @@ mraa_gpio_init(int pin)
         printf("gpio: pin %i beyond platform definition\n", pin);
         return NULL;
     }
+
+    struct device* pinmux_dev = device_get_binding(CONFIG_PINMUX_DEV_NAME);
+    if (pinmux_dev == NULL) {
+        printf("Failed to get binding for pinmux\n");
+    }
```
Comment (jontrulson) [inline source/gpio.c]: return NULL

File: source/i2c.c [hunk 3]
```diff
@@ -45,6 +46,20 @@ mraa_i2c_init(int bus)
         return NULL;
     }
 
+    struct device* pinmux_dev = device_get_binding(CONFIG_PINMUX_DEV_NAME);
+    if (pinmux_dev == NULL) {
+        printf("Failed to get binding for pinmux\n");
+    }
```
Comment (jontrulson) [inline source/i2c.c]: return NULL

File: source/pwm.c [hunk 4]
```diff
@@ -66,6 +66,28 @@ mraa_pwm_init(int pin)
     if (pin < 0 || pin >= board->phy_pin_count) {
         return NULL;
     }
+
+    struct device* pinmux_dev = device_get_binding(CONFIG_PINMUX_DEV_NAME);
+    if (pinmux_dev == NULL) {
+        printf("Failed to get binding for pinmux\n");
+    }
```
Comment (jontrulson) [inline source/pwm.c]: return NULL

File: source/uart.c [hunk 5]
```diff
@@ -49,6 +50,28 @@
...
+
+    if (board == NULL) {
+        printf("uart: platform not initialised\n");
+        return NULL;
+    }
+
+    struct device* pinmux_dev = device_get_binding(CONFIG_PINMUX_DEV_NAME);
+    if (pinmux_dev == NULL) {
+        printf("Failed to get binding for pinmux\n");
+    }
```
Comment (jontrulson) [inline source/uart.c]: return NULL

**Comment by malikabhi05** [comment]
@jontrulson thanks for pointing this out, will fix it

**Comment by arfoll** [comment]
Merged

**Pull request closed without merging**